### PR TITLE
New buildchain

### PR DESCRIPTION
**Pull request by nmellado**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Switch to modern cmake buildchain, with reliable installation and package generation procedure. 

* **What is the current behavior?** (You can also link to an open issue here)
Current buildchain is strongly hardcoded and does not provide relocable installation tools.

* **What is the new behavior (if this is a feature change)?**
Steps and their validation status:
   * Core, Engine, IO, PluginBase and GuiBase compille and can be installed by running `make install`
   * Dependencies:
        * source code is fetched and compiled at configuration time,
        * each library (e.g. Core, IO) manage its dependency indepandently.
   * Documentation compilation rules and structure has been updated, outdated pages marked.
   * Tests have been updated: now use Catch (unit tests) and ctest.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Yes, the documentation will be updated to describe how to compile Radium and link against its libraries,

* **Other information**:

### CODE REVIEW

**Comment by MathiasPaulin** [comment]
Can someone test this PR on Linux (and Windows), together with Radium-apps (branch fix-compilation) as I'm able to only test on MasOsX.
I've also a question : 
Not all the applications must be a Bundle on MacOs. 
Command line tools are not bundled, using a bundled application on command line is feasible but not quite user friendly.
So, following the actual proposed `configure_radium_app_install` cmake function,
we can propose a gui and a cmd-tool version of this function. On Linux, both will be the same, on MacOs, the former will build and install a bundle, the later will function as under Linux. For Windows ... I let you choose.

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Radium-Engine/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        REQUIRE( grey.isApprox( Color::Grey( 0.5 ) ) );
+    }
+    SECTION( "Test color conversion" ) {
+        Color grey2 = Color::fromRGB( Eigen::Matrix<Scalar, 3, 1>::Constant( 0.5 ) );
+
+        // Conversion from RBG failed
+        REQUIRE( grey.isApprox( grey2 ) );
+
+        Color grey3;
+        grey3 << .5, .5, .5, 1.;
```
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Radium-Engine/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Radium-Engine/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Radium-Engine/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 2]
```diff
@@ -83,43 +103,67 @@ FileData* TinyPlyFileLoader::loadFile( const std::string& filename ) {
...
     startTime = std::clock();
 
     file.read( ss );
 
-    geometry->setVertices( *(
-        reinterpret_cast<std::vector<Eigen::Matrix<float, 3, 1, Eigen::DontAlign>>*>( &verts ) ) );
-    geometry->setFrame( Core::Transform::Identity() );
+    {
+        const size_t numVerticesBytes = vertBuffer->buffer.size_bytes();
+        std::vector<Eigen::Matrix<float, 3, 1, Eigen::DontAlign>> verts( vertBuffer->count );
```
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 3]
```diff
@@ -83,43 +103,67 @@ FileData* TinyPlyFileLoader::loadFile( const std::string& filename ) {
...
+        std::memcpy( verts.data(), vertBuffer->buffer.get(), numVerticesBytes );
```
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 4]
```diff
@@ -83,43 +103,67 @@ FileData* TinyPlyFileLoader::loadFile( const std::string& filename ) {
...
     std::clock_t startTime;
     startTime = std::clock();
 
     file.read( ss );
 
-    geometry->setVertices( *(
-        reinterpret_cast<std::vector<Eigen::Matrix<float, 3, 1, Eigen::DontAlign>>*>( &verts ) ) );
-    geometry->setFrame( Core::Transform::Identity() );
+    {
+        const size_t numVerticesBytes = vertBuffer->buffer.size_bytes();
```
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
I confirm that it is not possible to move the folder Bundle-GNU on linux.
To reproduce:
 - compile radium and radium-apps
 - move bundle-gnu where you want
 - go to bundle-gnu/bin
 - run `./main-app`
In my case radiumCore cannot be found.

**Comment by MathiasPaulin** [comment]
> I confirm that it is not possible to move the folder Bundle-GNU on linux.

I'll give a look as soon as possible.

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Radium-Engine/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 2 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 3 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 4 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Radium-Engine/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 2 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 3 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 4 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 2 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 3 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 4 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 2 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 3 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 4 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 2 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 3 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 4 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470&bid=13735259)

File: tests/unittest/Core/color.cpp (hunk 1 above)
Comment (nmellado) [inline tests/unittest/Core/color.cpp]: Codacy Issue found: [Found suspicious operator ','](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 2 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 3 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp (hunk 4 above)
Comment (nmellado) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp]: Codacy Issue found: [Possible null pointer dereference: vertBuffer](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3960470)

**Comment by nmellado** [comment]
@MathiasPaulin I have made the changes we discussed today.
Could you please double check it is all good ?

**Pull request merged**